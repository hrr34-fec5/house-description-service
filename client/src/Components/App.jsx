import React from 'react';
import axios from 'axios';
import Overview from './overview.jsx';
import BelowReviews from './belowReviews.jsx';

class App extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      currentHouse: {}
    };
  }
  componentDidMount() {
    axios.get('/home', {
    params: {
      houseName: "Ducimus aut officiis qui voluptas rerum quaerat."
    }
  }).then(res => {
    let data = res.data;
    this.setState({currentHouse: data})
  });

  }
  render () {
    return (
      <div>
        <div>
          <Overview
          houseName={this.state.currentHouse.houseName}
          numberOfGuests={this.state.currentHouse.numberOfGuests}
          numberOfBedrooms={this.state.currentHouse.numberOfBedrooms}
          numberOfBeds={this.state.currentHouse.numberOfBeds}
          numberOfBaths={this.state.currentHouse.numberOfBaths}
          highlights={this.state.currentHouse.Highlights}
          description={this.state.currentHouse.description}
          />
        </div>
        <div>
          <BelowReviews
          neighborhood={this.state.currentHouse.Neighborhood}
          numberOfReviews={this.state.currentHouse.numberOfReviews}
          numberOfReferences={this.state.currentHouse.numberOfReferences}
          hostProfile={this.state.currentHouse.HostProfile} />
        </div>
      </div>
      );
  }
}


export default App;