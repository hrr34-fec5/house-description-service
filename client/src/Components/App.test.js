import React from 'react';
import { shallow } from 'enzyme';
import App from './App.jsx';
import Overview from './overview.jsx';
import BelowReviews from './belowReviews.jsx';

describe('Overview', () => {
  it('App should render correctly', () => {
    const component = shallow(<App/>);

    expect(component).toMatchSnapshot();
  });

  it('Overview should render correctly', () => {
    const component = shallow(<Overview/>);
    expect(component).toMatchSnapshot();
  });

  it('BelowReviews should render correctly', () => {
    const component = shallow(<BelowReviews/>);
    expect(component).toMatchSnapshot();
  });
});
/*
write tests be done by the end of the day simple tests
(test 70% of my lines of code.)
start working on proxy servers thursdays don't over complicate it.
look at the example and see what I am suppose to do in terms of a proxy server for this project
deploy on Friday with AWS.
final touches and conversations saturday and group video.
*/