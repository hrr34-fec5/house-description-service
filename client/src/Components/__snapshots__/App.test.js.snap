// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview App should render correctly 1`] = `
<div>
  <div>
    <Overview />
  </div>
  <div>
    <belowReviews />
  </div>
</div>
`;

exports[`Overview BelowReviews should render correctly 1`] = `
<div>
  <div
    className="description"
  >
    <div
      className="title"
    >
      The neighborhood
    </div>
     
    <br />
  </div>
  <div
    className="description"
  >
    <div
      className="title"
    >
      Hosted by Steve
      <br />
      <i
        className="fas fa-star"
      />
      <span
        className="icons"
      >
         Reviews
      </span>
      <i
        className="fas fa-user-check"
      />
      <span
        className="icons"
      >
         References
      </span>
    </div>
    <br />
  </div>
</div>
`;

exports[`Overview Overview should render correctly 1`] = `
<div>
  <h1
    className="header"
  />
  <div
    className="numberOf"
  >
    <i
      className="fas fa-users"
    />
    <span
      className="icons"
    >
       guests
    </span>
    <i
      className="fas fa-door-open"
    />
    <span
      className="icons"
    >
       bedrooms
    </span>
    <i
      className="fas fa-bed"
    />
    <span
      className="icons"
    >
       beds
    </span>
    <i
      className="fas fa-bath"
    />
    <span
      className="icons"
    >
       baths
    </span>
  </div>
  <div
    className="highlights"
  >
     
    <div
      className="title"
    >
      Highlights
    </div>
     
    <br />
  </div>
  <div
    className="description"
  >
    <div
      className="title"
    >
      The space
    </div>
     
    <br />
  </div>
  <div
    className="contact"
  >
    Contact host
  </div>
  <div
    className="border-line"
  />
  <div
    className="amenities"
  >
    <div
      className="title"
    >
      Amenities
    </div>
    <br />
    <i
      className="fas fa-laptop"
    />
    <span
      className="amenity-items"
    >
      Laptop friendly work space
    </span>
    <br />
    <i
      className="fas fa-wifi"
    />
    <span
      className="amenity-items"
    >
      Wifi
    </span>
    <br />
    <i
      className="fas fa-tv"
    />
    <span
      className="amenity-items"
    >
      TV
    </span>
    <br />
  </div>
</div>
`;
